### DISCUSSION TITLE
🥳  Web3Modal v3 now available 🥳

### DISCUSSION THREAD

**Discussion by chadyj** [WalletConnect/walletconnect-monorepo#3497] [Developer Support]
Web3Modal v3 is now available for early testing and review. 

- v3 docs and installation instructions can be found here https://docs.walletconnect.com/2.0/web3modal/v3/about
- A v3 demo is available at https://web3modal-laboratory-v3.vercel.app/

![v3](https://github.com/WalletConnect/walletconnect-monorepo/assets/66949816/51ea2b35-234f-467f-989f-5b6972ee3269)

v3 offers a new streamlined UI with several UX improvements. There are also many performance enhancements resulting in a smaller bundle size and better devex.

**Comment by softwarecurator** [comment]
would love to see base and base goerli support on this getting errors on those chains

**Reply by chadyj** [reply]
Thanks for the feedback!
Base and Base Goerli chains are newly added to v3.

It will be released in the next alpha this week.

**Comment by softwarecurator** [comment]
the simple look is so much better

**Comment by chadyj** [comment]
Alpha 4 has been released.  🎉 
Earlier alpha projects should upgrade due to a breaking change with the API.
See the full changelog https://github.com/WalletConnect/web3modal/releases/tag/3.0.0-alpha.4

**Reply by dirtycajunrice** [reply]
import { createWeb3Modal, defaultWagmiConfig } from '@web3modal/wagmi/react'

react is not an exported entrypoint of `@web3modal/wagmi`.

**Reply by xzilja** [reply]
@dirtycajunrice none of our team members are able to re-create this at the moment, could you create a codesandbox or something similar where issue can be replicated, will have a look.

**Reply by xzilja** [reply]
We managed to replicate on earlier typescript versions, fix is now available as part of `alpha.5`

**Comment by mikestead** [comment]
Hi, like the updates you've made for v3. A question regarding migration from v2. 

I keep track of registered wallets and use `showQrModal` option to trigger direct connection back to a previous wallet. After upgrading WC, the same option will trigger the QR modal, but it's skinned like v2. 

- Is there support for opening the QR modal with v3 skin using `WalletConnectConnector.connect({ showQrModal: true })`?
- Are v2 components + skin getting bundled with v3 still or perhaps that's some remanence of my previous config?

**Reply by GAEAlimited** [reply]
The integration and migration from WalletConnect v2 to v3 can involve changes in how certain features and components are handled. In WalletConnect v3, there might be differences in the appearance and behavior of certain UI elements, including the QR modal. Here are some considerations:

1. **QR Modal Styling**:
   - WalletConnect v3 may have introduced a new UI design, and the QR modal may now look different from v2.

2. **Configuration Options**:
   - The `showQrModal` option should still work in WalletConnect v3 to trigger the QR modal. However, the appearance of the modal may be controlled by the updated styling of v3.

3. **Component Bundling**:
   - It's possible that WalletConnect v3 bundles v2 components for compatibility reasons or because your previous configuration included v2 components. You may want to check if you have both v2 and v3 components in your project.

To ensure that the QR modal in WalletConnect v3 is styled according to the v3 design, you should:

- Verify that you are using the latest version of WalletConnect v3, as newer versions may have addressed styling issues.
- Review the documentation and release notes of WalletConnect v3 to see if there are any specific styling options or themes that can be applied.

If you are still encountering issues with the appearance of the QR modal in v3, you might consider reaching out to the WalletConnect development community or support channels for further assistance. They can provide guidance on how to adapt your integration to match the v3 design and behavior.

**Reply by xzilja** [reply]
I believe `WalletConnectConnector` bit is from wagmi? It doesn't use Web3Modal, instead it uses a more lightweight version called [WalletConnect Modal](https://docs.walletconnect.com/2.0/advanced/walletconnectmodal/about)

This one will remain as is for time being until we complete all tasks we set out for Web3Modal. Once that's done, WalletConnect modal will also get updated ui, but will mostly remain the same.

**Reply by mikestead** [reply]
thanks for clarification, makes sense

**Comment by kieranbop1990** [comment]
Do we have a date when non evm chains might be enabled?

**Reply by xzilja** [reply]
We've re-built v3 with this in mind, tl;dr we need to make an adapter similar to what we have for wagmi now, but using libraries of other chains i.e. solana web3. However there is no set deadline for this, we might try to partner up with projects on different chains and make this possible as a community effort as well.

**Comment by meri-maki** [comment]
Hi! Is there a way to show Recent transactions on Web3Modal (account button) 
it's present on https://web3modal.com/, but I cannot find any documentation on how to add it in react

**Comment by araf-hossain** [comment]
Hey @0xAsimetriq , 
I wanna integrate V3 using Vanilla JS. So, I am currently using [WalletConnect Modal](https://docs.walletconnect.com/advanced/walletconnectmodal/about). But it's not working with the Browser Extension. Do you have any suggestions for it, please? 
I am stuck in here around more than 2 weeks. You can have a look into here. [https://shop.bcl.xyz/collections/krapopolis](https://shop.bcl.xyz/collections/krapopolis)

**Reply by saintmaxi** [reply]
Hi, I know this is late but you want the Web3Modal, not WalletConnectModal, since WalletConnect Modal will rely on the QR code and WalletConnect to establish a session rather than potentially using the browser extension available. Here are the docs for Web3Modal https://docs.walletconnect.com/web3modal/javascript/about.

**Comment by phelix001** [comment]
You should limit namesize or you end up with this:
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/5187100/f6d82d2a-fa7b-49bb-81ba-a5e47671622f)

**Reply by phelix001** [reply]
or this 
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/5187100/c921fd19-16fd-4fdf-87e4-a7ff16269a67)

**Comment by taildfoobla** [comment]
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/142214404/15502d78-eabd-4a50-9e67-4689ae38d4e8)
Please i need help
I use web3modal in mobile to connect wallet but it stay here and nothing happen.
Is this a bug or how do i config this?
Thank you.

**Reply by taildfoobla** [reply]
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/142214404/b49fd97c-8d05-4cd3-9a7d-c6ed005f0769)
When i use public demo, it will pop like this
